web-common/src/features/dashboards/url-state/filters/convertors.ts

```diff
@@ -0,0 +1,69 @@
+import {
+  createAndExpression,
+  createBinaryExpression,
+  createInExpression,
+  createOrExpression,
+  createSubQueryExpression,
+  getAllIdentifiers,
+} from "@rilldata/web-common/features/dashboards/stores/filter-utils";
+import { V1Expression, V1Operation } from "@rilldata/web-common/runtime-client";
+import grammar from "./expression.js";
+import nearley from "nearley";
+
+export function convertFilterParamToExpression(filter: string) {
+  const parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar));
```

Comment:
Should we re-use a single parser instance, rather than creating a new parser instance every time we call this `convertFilterParamToExpression` function?

Reply:
It is meant to be a single use. We can reuse the compiled grammer.